cunumeric/array.py

```diff
@@ -150,7 +153,8 @@ def convert_to_cunumeric_ndarray(obj: Any, share: bool = False) -> ndarray:
         return obj
     # Ask the runtime to make a numpy thunk for this object
     thunk = runtime.get_numpy_thunk(obj, share=share)
-    return ndarray(shape=None, thunk=thunk)
+    flags = obj.flags if isinstance(obj, np.ndarray) and share else None
+    return ndarray(shape=None, thunk=thunk, flags=flags)
```

Comment:
```suggestion
    writeable = obj.flags.writeable if isinstance(obj, np.ndarray) and share else True
    return ndarray(shape=None, thunk=thunk, writeable=writeable)
```

At this point I don't think we even need to pass the flags from the NumPy array we're attaching to; the new cuNumeric array's `__array__()` will be set to that NumPy array automatically, so we'll just get all its flags directly through the `flagsobj` mechanism. We just need to propagate the source array's `.writeable` attribute, which is tracked separately (the signature for `ndarray.__init__` would need to change, along with the various places where we propagate `flags` when creating a view).

Reply:
It seems a little off to me to have an array param for both the entire flags object, and for one of the individual flags separately.

Reply:
https://github.com/nv-legate/cunumeric/pull/759/commits/07b436063293f3343cd7fd3013248d0044413328 (note I have a module func access `ndarray._writeable` internally... seemed better as implementation than adding public API that does not exist `np.ndarray`)